insecure/integration/tests/gossipsub/scoring/scoring_test.go

```diff
@@ -0,0 +1,154 @@
+	}{
+		{
+			name: "unknown peer, invalid signature",
+			spamMsgFactory: func(spammerId peer.ID, _ peer.ID, topic channels.Topic) *pubsub_pb.Message {
+				return p2ptest.PubsubMessageFixture(t, p2ptest.WithTopic(topic.String()))
+			},
+		},
+		{
+			name: "unknown peer, missing signature",
+			spamMsgFactory: func(spammerId peer.ID, _ peer.ID, topic channels.Topic) *pubsub_pb.Message {
+				return p2ptest.PubsubMessageFixture(t, p2ptest.WithTopic(topic.String()), p2ptest.WithoutSignature())
+			},
+		},
+		{
+			name: "known peer, invalid signature",
+			spamMsgFactory: func(spammerId peer.ID, _ peer.ID, topic channels.Topic) *pubsub_pb.Message {
+				return p2ptest.PubsubMessageFixture(t, p2ptest.WithFrom(spammerId), p2ptest.WithTopic(topic.String()))
+			},
+		},
+		{
+			name: "known peer, missing signature",
+			spamMsgFactory: func(spammerId peer.ID, _ peer.ID, topic channels.Topic) *pubsub_pb.Message {
+				return p2ptest.PubsubMessageFixture(t, p2ptest.WithFrom(spammerId), p2ptest.WithTopic(topic.String()), p2ptest.WithoutSignature())
+			},
+		},
+		{
+			name: "self-origin, invalid signature", // bounce back our own messages
+			spamMsgFactory: func(_ peer.ID, victimId peer.ID, topic channels.Topic) *pubsub_pb.Message {
+				return p2ptest.PubsubMessageFixture(t, p2ptest.WithFrom(victimId), p2ptest.WithTopic(topic.String()))
+			},
+		},
+		{
+			name: "self-origin, no signature", // bounce back our own messages
+			spamMsgFactory: func(_ peer.ID, victimId peer.ID, topic channels.Topic) *pubsub_pb.Message {
+				return p2ptest.PubsubMessageFixture(t, p2ptest.WithFrom(victimId), p2ptest.WithTopic(topic.String()), p2ptest.WithoutSignature())
+			},
+		},
+		{
+			name: "no sender",
+			spamMsgFactory: func(_ peer.ID, victimId peer.ID, topic channels.Topic) *pubsub_pb.Message {
+				return p2ptest.PubsubMessageFixture(t, p2ptest.WithoutSignerId(), p2ptest.WithTopic(topic.String()))
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			testGossipSubInvalidMessageDeliveryScoring(t, tc.spamMsgFactory)
+		})
+	}
+}
+
+// testGossipSubInvalidMessageDeliveryScoring tests that when a victim peer is spammed with invalid messages from
+// a spammer peer, the victim will eventually penalize the spammer and stop receiving messages from them.
+// Args:
+// - t: the test instance.
+// - spamMsgFactory: a function that creates unique invalid messages to spam the victim with.
+func testGossipSubInvalidMessageDeliveryScoring(t *testing.T, spamMsgFactory func(peer.ID, peer.ID, channels.Topic) *pubsub_pb.Message) {
+	role := flow.RoleConsensus
+	sporkId := unittest.IdentifierFixture()
+	blockTopic := channels.TopicFromChannel(channels.PushBlocks, sporkId)
+
+	idProvider := mock.NewIdentityProvider(t)
+	spammer := corruptlibp2p.NewGossipSubRouterSpammer(t, sporkId, role, idProvider)
+	ctx, cancel := context.WithCancel(context.Background())
+	signalerCtx := irrecoverable.NewMockSignalerContext(t, ctx)
```

Comment:
Since we're using mocks, that's another reason not to use the word `integration` for the path of this test file. Integration test to me implies no mocks.

Reply:
Added the following note as the command to the code: 
```
// Note: the term integration is used here because it requires integrating all components of the libp2p stack.
```